src/code.ts

```diff
@@ -1,45 +1,39 @@
-import type { Document } from './bson';
+import type { BSONDocument } from './bson';
 
-/**
- * A class representation of the BSON Code type.
- */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type CodeFunction = (...args: any[]) => any;
```

Comment:
Is there something confusing about this type?
We can Function just throws warnings which I can silence, maybe even globally in the config but if not any funciton that works with this type will have to be annotated. This accepts any function.

Reply:
Hm, I don't know where my comment to this went.. I'm wondering because in spirit the `Code` type is meant to be a wrapper around a JavaScript function, so it seemed weird to me that we would introduce a new type which accepts any function rather than just using the primitive type.

Maybe we can disable the `Function` rule just for the files where we need to use it? When it comes to BSON there will have to be exceptions, we already disable linter rules for `any` in a number of key places because.. well you need to support `any` sometimes! I'd rather not introduce new types just to get around a linter warning, if it's going to make the code less readable